subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherUpdater.java

```diff
@@ -25,27 +25,28 @@
 
 public interface FileWatcherUpdater extends SnapshotHierarchy.SnapshotDiffListener {
     /**
-     * Changes the root project directories, e.g. when the same daemon is used on a different project.
+     * Registers a hierarchy to watch.
      *
-     * The root project directories are used by hierarchical watchers to minimize the number of watched roots
-     * by watching the root projects instead of watching directories inside.
+     * The watcher registry will only watch for changes in hierarchies to watch.
      *
      * @throws WatchingNotSupportedException when the native watchers can't be updated.
      */
-    void updateRootProjectDirectories(Collection<File> updatedRootProjectDirectories);
+    void registerHierarchyToWatch(File hierarchyToWatch, SnapshotHierarchy root);
```

Comment:
`ToWatch` seems to imply to me that we'll be starting to watch the thing as the result of this call. Maybe `registerWatchableHierarchies()` is a better name? I don't like "watchable" as a word, but it does capture the conditional nature of the thing.

Reply:
I am using now watchable hierarchy.